crates/core_simd/src/masks/to_bitmask.rs

```diff
@@ -30,19 +30,18 @@ pub trait ToBitMask: Sealed {
 /// Converts masks to and from byte array bitmasks.
 ///
 /// Each bit of the bitmask corresponds to a mask lane, starting with the LSB of the first byte.
-#[cfg(feature = "generic_const_exprs")]
 pub trait ToBitMaskArray: Sealed {
-    /// The length of the bitmask array.
-    const BYTES: usize;
+    /// The bitmask array.
+    type BitMaskArray;
```

Comment:
this should have some useful bounds, such as `Copy + Unpin + Send + Sync + 'static + AsRef<[u8]> + AsMut<[u8]> + BorrowMut<[u8]>` and maybe more

Reply:
Would adding more bounds in the future be backwards compatible?

Reply:
No, bounds can only be relaxed forward-compatibly, and only sometimes.

Reply:
Is that still true when the trait is sealed?

Reply:
...that does change things a bit, since the major compat hazard is people referencing the type generically.